optimum/intel/openvino/modeling_base.py

```diff
@@ -229,6 +236,20 @@ def _create_infer_request(self, model):
         compiled_model = core.compile_model(model, self._device, self.ov_config)
         return compiled_model.create_infer_request()
 
+    def _reshape(self, model: openvino.runtime.Model, batch_size: int, sequence_length: int):
+        shapes = {}
+        for inputs in model.inputs:
+            shapes[inputs] = inputs.get_partial_shape()
+            shapes[inputs][0] = batch_size
+            shapes[inputs][1] = sequence_length
+        model.reshape(shapes)
+        return model
+
+    def reshape(self, batch_size: int, sequence_length: int):
```

Comment:
The current implementation works the way that the user needs to reshape the model every time exactly the shape required for the input. So that if I reshape the model to [1,32] and pass the input with [1,17], I got an error. My assumption was that it should be possible to define the static shape (max sequence length) and if the user passes the input which is less than the shape it will be padded and the output will be truncated.

Reply:
I also tried a w/a proposed by @helena-intel ```tokenizer = AutoTokenizer.from_pretrained(model_checkpoint, max_length=32, padding="max_length")``` but it does not work.

Reply:
Can you try :
```python
inputs = tokenizer(text, return_tensors="pt", max_length=32, padding="max_length", truncation=True)
```